packages/react/src/Banner/Banner.tsx

```diff
@@ -0,0 +1,360 @@
+}
+
+const iconForVariant: Record<BannerVariant, React.ReactNode> = {
+  critical: <StopIcon />,
+  info: <InfoIcon />,
+  success: <CheckCircleIcon />,
+  upsell: <InfoIcon />,
+  warning: <AlertIcon />,
+}
+
+export const Banner = React.forwardRef<HTMLElement, BannerProps>(function Banner(
+  {children, description, icon, onDismiss, primaryAction, secondaryAction, title, variant = 'info', ...rest},
+  ref,
+) {
+  const titleId = useId()
+  const value = useMemo(() => {
+    return {
+      titleId,
+    }
+  }, [titleId])
+  const dismissible = variant !== 'critical' && onDismiss
+  const hasActions = primaryAction || secondaryAction
+
+  if (__DEV__) {
+    // Note: __DEV__ will make it so that this hook is consistently called, or
+    // not called, depending on environment
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useEffect(() => {
+      const title = document.getElementById(titleId)
+      if (!title) {
+        throw new Error(
+          'The Banner component requires a title to be provided as the `title` prop or through `Banner.Title`',
+        )
+      }
+    }, [titleId])
+  }
+
+  return (
+    <BannerContext.Provider value={value}>
+      <StyledBanner {...rest} aria-labelledby={titleId} as="section" data-variant={variant} tabIndex={-1} ref={ref}>
+        <style>{BannerContainerQuery}</style>
+        <div className="BannerIcon">{icon && variant === 'info' ? icon : iconForVariant[variant]}</div>
+        <div className="BannerContainer">
+          <div className="BannerContent">
+            {title ? <BannerTitle>{title}</BannerTitle> : null}
+            {description ? <BannerDescription>{description}</BannerDescription> : null}
+            {children}
+          </div>
+          {hasActions ? <BannerActions primaryAction={primaryAction} secondaryAction={secondaryAction} /> : null}
+        </div>
+        {dismissible ? (
+          <IconButton
+            aria-label="Dismiss banner"
+            onClick={onDismiss}
+            className="BannerDismiss"
+            icon={XIcon}
+            variant="invisible"
+          />
+        ) : null}
+      </StyledBanner>
+    </BannerContext.Provider>
+  )
+})
+
+/**
+ * For styling, it's important that the icons and the text have the same height
+ * for alignment to occur in multi-line scenarios. Currently, we use a
+ * line-height of `20px` so that means that the height of icons should match
+ * that value.
+ */
+const StyledBanner = styled.div`
+  display: grid;
+  grid-template-columns: auto minmax(0, 1fr) auto;
+  align-items: start;
+  background-color: var(--banner-bgColor);
+  border: var(--borderWidth-thin, 1px) solid var(--banner-borderColor);
+  padding: var(--base-size-8, 0.5rem);
+  border-radius: var(--borderRadius-medium, ${get('radii.2')});
+
+  @supports (container-type: inline-size) {
+    container: banner / inline-size;
+  }
+
+  &[data-variant='critical'] {
+    --banner-bgColor: ${get('colors.danger.subtle')};
```

Comment:
@joshblack Can we use v8 tokens instead and avoid `get` all together?

```suggestion
    --banner-bgColor: var(--bgColor-danger-muted);
```

Reply:
@lukasoppermann sadly not yet 😞 this will likely be a part of v37 when we officially shift over.